runtime-testsuite/test/org/antlr/v4/test/runtime/csharp/BaseCSharpTest.java

```diff
@@ -493,83 +493,55 @@ public boolean createProject() {
-                if (!thisSource.isDirectory()) {
-                    File thisDest = new File(tmpdir + "/src/" + thisSource.getName());
-                    boolean success = thisSource.renameTo(thisDest);
-
-                    if (!success) {
-                        throw new RuntimeException("Moving file " + thisSource + " to " + thisDest + " failed.");
-                    }
-                }
-            }
-
             // save auxiliary files
             String pack = BaseCSharpTest.class.getPackage().getName().replace(".", "/") + "/";
-            saveResourceAsFile(pack + "global.json", new File(tmpdir, "global.json"));
-            saveResourceAsFile(pack + "project.json", new File(tmpdir, "src/project.json"));
-            return true;
-        }
-        catch(Exception e) {
-            e.printStackTrace(System.err);
-            return false;
-        }
-    }
-
-    public boolean buildDotnetProject() {
-        // find runtime package
-        final ClassLoader loader = Thread.currentThread().getContextClassLoader();
-        final URL runtimeProj = loader.getResource("CSharp/runtime/CSharp/Antlr4.Runtime/Antlr4.Runtime.dotnet.xproj");
-        if ( runtimeProj==null ) {
-            throw new RuntimeException("C# runtime project file not found!");
-        }
-        File runtimeProjFile = new File(runtimeProj.getFile());
-        String runtimeProjPath = runtimeProjFile.getParentFile().getParentFile().getPath();
-        String projectRefPath = runtimeProjPath.substring(0, runtimeProjPath.lastIndexOf("/"));
+            saveResourceAsFile(pack + "Antlr4.Test.dotnet.csproj", new File(tmpdir, "Antlr4.Test.dotnet.csproj"));
 
-        // update global.json to reference runtime path
-        try {
-            String content = new java.util.Scanner(new File(tmpdir + "/global.json")).useDelimiter("\\Z").next();
-            content = content.replaceAll("replace_this", projectRefPath);
-            java.io.PrintWriter out = new java.io.PrintWriter(tmpdir + "/global.json");
-            out.write(content);
-            out.close();
-        }
-        catch(Exception e) {
-            return false;
-        }
-
-        // build test
-        String dotnetcli = locateTool("dotnet");
-        String[] args = new String[] {
-            dotnetcli,
-            "restore",
-            ".",
-            projectRefPath
-        };
+            // find runtime package
+            final ClassLoader loader = Thread.currentThread().getContextClassLoader();
+            final URL runtimeProj = loader.getResource("CSharp/runtime/CSharp/Antlr4.Runtime/Antlr4.Runtime.dotnet.csproj");
+            if ( runtimeProj==null ) {
+                throw new RuntimeException("C# runtime project file not found!");
+            }
+            File runtimeProjFile = new File(runtimeProj.getFile());
+            String runtimeProjPath = runtimeProjFile.getPath();
 
-        try {
+            // add Runtime project reference
```

Comment:
Adding a reference to the project (runtimeProjPath) rather than the Assembly is likely to be the cause of the very long Travis build with both Linux and MacOSX. I'm guessing this tells dotnet to rebuild both the test project and the runtime project.
Have you tried simply referencing the Assembly here?